DevCycle/Models/EventQueue.swift

```diff
@@ -36,19 +36,26 @@ class EventQueue {
             return
         }
         
-        var eventsToFlush: [DVCEvent] = []
+        var eventsToFlush: [DVCRequestEvent] = []
         eventDispatchQueue.sync {
             self.flushing = true
-            eventsToFlush = self.events
-            eventsToFlush.append(contentsOf: aggregateEventQueue.getAllAggregateEvents())
+            eventsToFlush = self.events.map { (trackedEvent: DVCEvent) -> DVCRequestEvent in
+                return DVCRequestEvent.fromDVCEvent(dvcEvent: trackedEvent)
+            }
+            eventsToFlush.append(contentsOf: aggregateEventQueue.getAllAggregateEvents().map({
+                (variableEvent: DVCEvent) -> DVCRequestEvent in
+                    return DVCRequestEvent.fromDVCEvent(dvcEvent: variableEvent)
+            }))
             self.clear()
         }
         Log.debug("Flushing events: \(eventsToFlush.count)")
         service.publishEvents(events: eventsToFlush, user: user, completion: { data, response, error in
             
             if let error = error {
                 Log.error("Error: \(error)", tags: ["events", "flush"])
-                self.queue(eventsToFlush)
+                self.queue(eventsToFlush.map({ (requestEvent: DVCRequestEvent) -> DVCEvent in
+                    return DVCRequestEvent.toDVCEvent(dvcRequestEvent: requestEvent)
```

Comment:
do we actually need to convert it back? why not just leave it as a `DVCRequestEvent` queue

Reply:
I can change all instances of the queue to `DVCRequestEvent` then, I think that'll be better and eliminate the need for multiple conversions 🤔  might need to add a few more `queue` functions to support queuing either incoming `DVCEvent` or `DVCRequestEvent` types?

Reply:
or you can follow @ajwootto 's advice and convert it at the end before publishing https://github.com/DevCycleHQ/ios-client-sdk/pull/58#discussion_r866161961